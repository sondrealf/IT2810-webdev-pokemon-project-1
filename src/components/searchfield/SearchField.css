.searchfield {
  background-color: #fff;
  color: #000000;
  border: 0;
  border-radius: 5px 0px 0px 5px;
  padding: 0 1.5em;
  width: 100%;
  outline: none;
  font-size: 16px;
  height: 100%;
}

.searchfield__container {
  margin-top: 1.5em;
  display: flex;
  width: 100%;
  height: 3em;
}

.searchfield__button {
  padding: 1em;
  border-radius: 0px 5px 5px 0px;
  background-color: #db4032;
  border: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  transition: 200ms;
  width: 22px;
  height: 22px;
}

.searchfield__button:hover {
  background-color: hsl(5, 70%, 60%);
}

.searchfield__image {
  height: 50%;
}
