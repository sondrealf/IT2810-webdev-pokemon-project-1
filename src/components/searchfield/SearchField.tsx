import { useState } from "react";
import "./SearchField.css";
import { useNavigate } from "react-router-dom";
import SearchIcon from "../../assets/search.png";

/**
 * A component that renders a search field for searching Pokemon by name.
 * @returns {JSX.Element} - The SearchField component.
 */
const SearchField = () => {
  const [searchText, setSearchText] = useState("");
  const navigate = useNavigate();

  return (
    <form
      className="searchfield__container"
      onSubmit={() => navigate("/pokemon/" + searchText.toLowerCase())}
    >
      <input
        type="search"
        placeholder="Pokemon name"
        className="searchfield"
        value={searchText}
        onChange={(e) => setSearchText(e.target.value)}
      />
      <input type="image" src={SearchIcon} className="searchfield__button" />
    </form>
  );
};
export default SearchField;
