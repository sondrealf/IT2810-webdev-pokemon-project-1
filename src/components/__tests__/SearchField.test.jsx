import { render, screen } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { fireEvent } from "@testing-library/dom";
import SearchField from "../searchfield/SearchField";

const mockUseNavigate = vi.fn();

vi.mock("react-router-dom", async () => {
  const actual = await vi.importActual("react-router-dom");
  return {
    ...actual,
    useNavigate: () => mockUseNavigate,
  };
});

describe("SearchField", () => {
  test("matches snapshot", () => {
    const { container } = render(<SearchField />);
    expect(container).toMatchSnapshot();
  });

  test("should render searchfield", () => {
    render(<SearchField />);
    expect(screen.getByRole("searchbox")).toBeInTheDocument();
    expect(screen.getByRole("searchbox")).toHaveValue("");
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("should change textfield value", () => {
    render(<SearchField />);
    const search = screen.getByRole("searchbox");
    fireEvent.change(search, { target: { value: "pikachu" } });
    expect(search).toHaveValue("pikachu");
  });

  test("Pokemon page for the searched pokemon is shown on search submit", () => {
    render(<SearchField />);
    const search = screen.getByRole("searchbox");
    fireEvent.change(search, { target: { value: "vulpix" } });
    fireEvent.submit(screen.getByRole("button"));
    expect(mockUseNavigate).toHaveBeenCalledWith("/pokemon/vulpix");
  });

  test("Case sensitive search", () => {
    mockUseNavigate.mockClear();
    render(<SearchField />);
    const search = screen.getByRole("searchbox");
    fireEvent.change(search, { target: { value: "Vulpix" } });
    fireEvent.submit(screen.getByRole("button"));
    expect(mockUseNavigate).toHaveBeenCalledWith("/pokemon/vulpix");
  });
});
