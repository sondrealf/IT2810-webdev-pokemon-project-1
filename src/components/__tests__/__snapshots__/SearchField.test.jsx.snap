// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchField > matches snapshot 1`] = `
<div>
  <form
    class="searchfield__container"
  >
    <input
      class="searchfield"
      placeholder="Pokemon name"
      type="search"
      value=""
    />
    <input
      class="searchfield__button"
      src="/project1/src/assets/search.png"
      type="image"
    />
  </form>
</div>
`;
